### PR TITLE
Allow decoding embedded struct pointer in decodeMapFromStruct

### PR DESCRIPTION
**Pull request by guscott**
This may address #156 , but it is not my use case, so I did not validated this case particularly.

### CODE REVIEW

**Comment by mitchellh** [comment]
This looks good, but do you mind pulling this out into a separate standalone test case vs. overloading another? That makes it easier to remember what the test is addressing.

**Comment by guscott** [comment]
A new test case was added just for this.

File: mapstructure_test.go [hunk 1]
```diff
@@ -2316,17 +2351,12 @@ func TestDecode_StructTaggedWithOmitempty_KeepNonEmptyValues(t *testing.T) {
 		VisibleMapField:    nil,
 		OmitMapField:       map[string]interface{}{"k": "v"},
 		NestedField:        nil,
-		OmitNestedField:    &Nested{},
+		OmitNestedField:    nil,
```
Comment (codecat) [inline mapstructure_test.go:2354]: Was this change intended? Looks a bit suspicious to me.

Comment (guscott) [reply mapstructure_test.go:2354]: Yes, this is intended. This PR change the behavior of struct pointer to allow decoding them. This means that &Nested{} is no longer considered `nil` once decoded; it will be a empty `map[string]interface{}`. Since this is not what is tested here, a separate unit test (`TestNestedTypeStructPointer`) was added just for this and this test was modified to pass `nil` so that it is not overloaded.

**Comment by mitchellh** [comment]
Fixed by #205
